packages/salesforcedx-utils-vscode/test/unit/cli/deployResultParser.test.ts

```diff
@@ -0,0 +1,227 @@
+
+  beforeEach(() => {
+    deployErrorResult = {
+      message: 'Deploy failed.',
+      name: 'DeployFailed',
+      stack: '123',
+      status: 1,
+      warnings: [],
+      result: []
+    };
+    deploySuccessResult = {
+      status: 0,
+      result: { deployedSource: [] }
+    };
+  });
+
+  it('Should parse error info successfully', async () => {
+    const resultItem = {
+      filePath: 'src/apexclasses/Testing.cls',
+      error: 'asdf',
+      lineNumber: '1',
+      columnNumber: '1',
+      type: 'ApexClass',
+      fullName: 'Testing'
+    };
+
+    deployErrorResult.result.push(resultItem);
+
+    const parser = new ForceDeployResultParser(
+      JSON.stringify(deployErrorResult)
+    );
+    const errs = parser.getErrors();
+    if (errs) {
+      expect(errs.message).to.be.equals(deployErrorResult.message);
+      expect(errs.name).to.be.equals(deployErrorResult.name);
+      expect(errs.result)
+        .to.be.an('array')
+        .to.have.lengthOf(1);
+      expect(errs.result[0]).to.deep.equals(resultItem);
+      expect(errs.stack).to.be.equals(deployErrorResult.stack);
+      expect(errs.status).to.be.equals(deployErrorResult.status);
+      expect(errs.warnings).to.deep.equals(deployErrorResult.warnings);
+    } else {
+      throw Error('Errors should be present but were not returned');
+    }
+  });
+
+  it('Should parse incomplete error info successfully', async () => {
+    const stdOut = {
+      message:
+        'The DocumentFolder named folder/image.png was not found in the workspace.',
+      status: 1,
+      stack:
+        'SourceElementDoesNotExist: The DocumentFolder named folder/image.png was not found in the workspace.\n    at Function.create (some/internal/error.js:146:16)\n    at <anonymous>',
+      name: 'SourceElementDoesNotExist',
+      warnings: ['Some warning message from sfdx cli.']
+    };
+
+    const parser = new ForceDeployResultParser(JSON.stringify(stdOut));
+    const errs = parser.getErrors();
+    if (errs) {
+      expect(errs.message).to.be.equals(stdOut.message);
+      expect(errs.name).to.be.equals(stdOut.name);
+      expect(errs).to.not.have.property('result');
+      expect(errs.stack).to.be.equals(stdOut.stack);
+      expect(errs.status).to.be.equals(stdOut.status);
+      expect(errs.warnings).to.deep.equals(stdOut.warnings);
+    } else {
+      throw Error('Errors should be present but were not returned');
+    }
+  });
+
+  it('Should properly parse stdOut amongst output that needs to be ignored', async () => {
+    deployErrorResult.result.push({
+      filePath: 'src/apexclasses/Testing.cls',
+      error: 'Invalid dependency ...',
+      lineNumber: '10',
+      columnNumber: '23',
+      type: 'ApexClass',
+      fullName: 'Testing'
+    });
+
+    const parser = new ForceDeployResultParser(
+      `sfdx force:source:deploy --json --loglevel fatal --manifest /Users/username/manifest/package.xml ${
+        require('os').EOL
```

Comment:
If possible, lets change this to a top level import instead of an inline require.

Reply:
Bleh, i didn't notice it was in a test too. Thanks for catching